drivers/serial/uart_nrfx_uarte.c

```diff
@@ -252,6 +261,8 @@ struct uarte_nrfx_data {
 	(baudrate) == 921600 ? NRF_UARTE_BAUDRATE_921600 :	\
 	(baudrate) == 1000000 ? NRF_UARTE_BAUDRATE_1000000 : 0)
 
+#define LOW_POWER_ENABLED(_config) \
+	(IS_ENABLED(UARTE_ANY_LOW_POWER) && (_config->flags & UARTE_CFG_FLAG_LOW_POWER))
```

Comment:
Since the old low-power mechanism is not to be used with `PM_DEVICE`, maybe `!IS_ENABLED(CONFIG_PM_DEVICE)` should be added here, so that all related code could be optimized out?

Reply:
added